test/e2e/performance_test.go

```diff
@@ -0,0 +1,365 @@
+		b.Fatalf("Checking network policies realization failed: %v", err)
+	} else {
+		b.Log("Network policy realized")
+	}
+
+	serverURL := &url.URL{Scheme: "http", Host: nginxPodIP, Path: "/"}
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+		b.Logf("Running http request bench %d/%d", i+1, b.N)
+		cmd := []string{"ab", "-n", fmt.Sprint(times), "-c", fmt.Sprint(*httpConcurrency), serverURL.String()}
+		if stdout, stderr, err := data.runCommandFromPod(testNamespace, benchABPodName, abContainerName, cmd); err != nil {
+			b.Errorf("Error when running http request %dx: %v, stdout: %s, stderr: %s\n", times, err, stdout, stderr)
+		}
+	}
+}
+
+// networkPolicyRealize runs the benchmark of time cost of a network policy with `workloads` amount CIDRs
+// to be realized into flow entries. In order to have entities for the network policy to apply to, we
+// create two dummy Pods: apache-bench and Nginx, they don't have activity during the benchmark test.
+func networkPolicyRealize(workloads int, data *TestData, b *testing.B) {
+	setupPerformanceTestPods(data, b)
+	for i := 0; i < b.N; i++ {
+		go func() {
+			if err := populateWorkload(generateWorkload(workloads), data); err != nil {
+				b.Fatalf("Error when populating workload: %v", err)
+			}
+		}()
+
+		b.StartTimer()
+		if err := waitNetworkPolicyRealize(workloads, data); err != nil {
+			b.Fatalf("Checking network policies realization failed: %v", err)
+		} else {
+			b.Log("Network policy realized")
+		}
+		b.StopTimer()
+
+		if err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Delete(workloadNetworkPolicyName, new(metav1.DeleteOptions)); err != nil {
+			b.Fatalf("Error when cleaning up network policies after running one bench iteration: %v", err)
+		}
+	}
+}
+
+func dumpFlows(data *TestData) (string, error) {
+	antreaPodName, err := data.getAntreaPodOnNode(masterNodeName())
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{"ovs-ofctl", "dump-flows", "br-int"}
+	stdout, _, err := data.runCommandFromPod(AntreaNamespace, antreaPodName, "antrea-agent", cmd)
+	if err != nil {
+		return "", err
+	}
+	return stdout, nil
+}
+
+func waitNetworkPolicyRealize(workloads int, data *TestData) error {
+	done := make(chan struct{})
+	return wait.WaitFor(
+		func(stopCh <-chan struct{}) <-chan struct{} {
+			checkCh := make(chan struct{})
+			timeout := time.After(*realizeTimeout)
+			go func() {
+				defer close(done)
+				for {
+					select {
+					case <-stopCh: // realized
+						break
+					case <-timeout: // timeout
+						break
+					case checkCh <- struct{}{}: // signal next check
+					}
+				}
+			}()
+			return checkCh
+		},
+		func() (bool, error) { return checkRealize(workloads, data) },
+		done,
+	)
+}
+
+func checkRealize(workloads int, data *TestData) (bool, error) {
+	flowNums, err := countFlows(data)
+	if err != nil {
+		return false, fmt.Errorf("dumping flow keeps failed")
+	}
+	return flowNums > workloads, nil
```

Comment:
maybe it is worth adding a comment here:
 * it seems that we are ignoring non-policy flows here, presumably because there is a small number of them?
 * it seems that you make the assumption that because of the policy specification we are using for the test, each CIDR in the spec corresponds to one flow. That may not be obvious to someone reading the code if they are not familiar with the internals of Antrea's network policy implementation.

Reply:
Updated, PTAL.